### PR TITLE
Register only one `DOMContentLoaded` event listener in `onDOMContentLoaded`

### LINKED ISSUE #34103: Remove duplicate `DOMContentLoaded` events
Is this normal?

Captura de tela de 2021-05-14 00-04-56

Bootstrap 5.0.0

_Originally posted by @o-alquimista in https://github.com/twbs/bootstrap/discussions/33980_

### PR DESCRIPTION
**Pull request by alpadev**
Closes #34103 

Since #34114 is more focused around changing the way we register our components with jQuery, this PR is changing the `onDOMContentLoaded` utility function, to make use of an array to cache callbacks (while the document is loading) and execute them all at once. Therefore we register only one `DOMContentLoaded` event listener instead of one for every call to the function.

### CODE REVIEW

**Comment by GeoSot** [comment]
Seems fine by me

**Review by GeoSot — APPROVED** [review]

**Comment by alpadev** [comment]
@GeoSot Changed the implementation so it wouldn't add an event listener when the function isn't called during `loading` state. Updated the test to check that `addEventListener` isn't called more than once.

**Review by GeoSot — APPROVED** [review]

**Comment by GeoSot** [comment]
@rohit2sharma95  give it a look, if you like, for a second opinion

**Review by rohit2sharma95 — APPROVED** [review]
Good work @alpadev :1st_place_medal:

File: js/tests/unit/util/index.spec.js [hunk 1]
```diff
@@ -608,15 +608,24 @@ describe('Util', () => {
...
       const spy = jasmine.createSpy()
+      const spy2 = jasmine.createSpy()
+
+      spyOn(document, 'addEventListener').and.callThrough()
       spyOnProperty(document, 'readyState').and.returnValue('loading')
+
       Util.onDOMContentLoaded(spy)
+      Util.onDOMContentLoaded(spy2)
+
       window.document.dispatchEvent(new Event('DOMContentLoaded', {
```
[the remark this answers is not in the document]
Comment (alpadev) [reply js/tests/unit/util/index.spec.js]: Goot catch!

Me neither, so I just updated it. 👍

**Comment by alpadev** [comment]
@XhmikosR removed the IIFE as requested

**Review by mdo — APPROVED** [review]

**Pull request merged**